adhocracy4/reports/static/reports/ReportModal.jsx

```diff
@@ -1,89 +1,67 @@
 
-  handleTextChange (e) {
-    this.setState({ report: e.target.value })
-  }
+  const partials = {}
 
-  resetModal () {
-    if (!$('#' + this.props.name).is(':visible')) {
-      this.setState({
-        report: '',
-        showSuccessMessage: false,
-        showErrorMessage: false,
-        showReportForm: true,
-        errors: null
-      })
-    } else {
-      setTimeout(this.resetModal.bind(this), 500)
-    }
+  const handleTextChange = (e) => {
+    setReport(e.target.value)
   }
 
-  submitReport () {
+  const submitReport = () => {
     api.report.submit({
-      description: this.state.report,
-      content_type: this.props.contentType,
-      object_pk: this.props.objectId
+      description: report,
+      content_type: props.contentType,
+      object_pk: props.objectId
     })
-      .done(function () {
-        this.setState({
-          report: '',
-          showSuccessMessage: true,
-          showReportForm: false,
-          showErrorMessage: false
-        })
-      }.bind(this))
+      .done(() => {
+        setReport('')
+        setShowSuccessMessage(true)
+        setShowReportForm(false)
+      })
   }
 
-  render () {
-    const thankyouText = django.gettext('Thank you! We are taking care of it.')
-    const placeholderText = django.gettext('Your message')
-    const sendReportTag = django.gettext('Send Report')
-    const partials = {}
-    if (this.state.showSuccessMessage) {
-      partials.title = (<span><i className="fa fa-check" /> {thankyouText}</span>)
-      partials.hideFooter = true
-      partials.bodyClass = 'success'
-    } else if (this.state.showReportForm) {
-      partials.title = this.props.title
-      partials.body = (
-        <div className="form-group">
-          <textarea
-            rows="5" className="form-control report-message" value={this.state.report}
-            placeholder={placeholderText} onChange={this.handleTextChange.bind(this)}
-          />
-          {this.state.errors && <span className="help-block">{this.state.errors.description}</span>}
-        </div>
-      )
-    }
-
-    return (
-      <Modal
-        abort={this.props.abort}
-        name={this.props.name}
-        handleSubmit={this.submitReport.bind(this)}
-        action={sendReportTag}
-        partials={partials}
-        dismissOnSubmit={false}
-      />
+  if (showSuccessMessage) {
+    partials.body = (
+      <div className="u-spacer-bottom-triple">
+        <i className="fa fa-check" /> {translations.thankyouText}
+      </div>
+    )
+    partials.hideFooter = true
+    partials.bodyClass = 'success'
+  } else if (showReportForm) {
+    partials.title = translations.sendReportTag
+    partials.description = props.description
+    partials.body = (
+      <div className="form-group">
+        <textarea
+          rows="5" className="form-control report-message" value={report}
+          placeholder={translations.placeholderText} onChange={handleTextChange}
+        />
+      </div>
```

Comment:
@phillimorland i was about to add this back again and now remember why I removed it. I did because `this.state.errors` never gets `set`. So this part never renders as `this.state.errors` is always `null`.

I am not sure what kind of errors this should show? do we care right now?

Reply:
@khamui ahh ok thats a problem, we should add an issue then